src/dali_backend.cc

```diff
@@ -178,7 +200,25 @@ class DaliModelInstance : public ::triton::backend::BackendModelInstance {
     dali_executor_ = std::make_unique<DaliExecutor>(std::move(pipeline));
   }
 
-
+  std::vector<IDescr> GenerateInputs(TritonRequest& request) {
+    uint32_t input_cnt = request.InputCount();
+    std::vector<IDescr> ret;
+    for (uint32_t input_idx = 0; input_idx < input_cnt; ++input_idx) {
+      auto input = request.InputByIdx(input_idx);
+      auto input_byte_size = input.ByteSize();
+      auto input_buffer_count = input.BufferCount();
+      std::vector<IBufferDescr> buffers;
+      for (uint32_t buffer_idx = 0; buffer_idx < input_buffer_count; ++buffer_idx) {
+        auto buffer = input.GetBuffer(buffer_idx);
+        ENFORCE(buffer.device == device_type_t::CPU || buffer.device_id == device_id_,
+                "GPU input must reside on the same device that the model instance.");
+        buffers.push_back(buffer);
+      }
+      ret.push_back({input.Meta(), buffers});
```

Comment:
Would this make any difference?
```suggestion
      ret.push_back({input.Meta(), std::move(buffers)});
```

Reply:
It might. Done